modules/ding_user/ding_user.module

```diff
@@ -243,6 +271,77 @@ function ding_user_form_alter(&$form, $form_state, $form_id) {
       array_splice($form['#validate'], $pos, 0, 'ding_user_user_login_validate');
+
+      // Add introduction text to the form.
+      $form['intro_text'] = array(
+        '#type' => 'container',
+        '#attributes' => array('class' => array('intro-text')),
+        'lead' => array(
+          '#type' => 'container',
+          '#attributes' => array('class' => array('lead')),
+          array('#markup' => t('If you want to see what you have borrowed, reserve books or edit your user profile, you must be logged in.')),
+        ),
+        'text' => array(
+          '#type' => 'container',
+          '#attributes' => array('class' => array('text')),
+        ),
+      );
+
+      $user_signup_link = variable_get('ding_user_signup_link', '');
+      if (!empty($user_signup_link)) {
+        $form['intro_text']['text'] = array(
+          '#markup' => t('If you are not a registered user, you can register !link, or you can sign up in person at your local library.', array(
+            '!link' => l(t('here'), $user_signup_link, array('external' => TRUE)),
+          )),
+        );
+      }
+      else {
+        $form['intro_text']['text'] = array(
+          '#markup' => t('If you are not a registered user, you can sign up in person at your local library.'),
+        );
+      }
+
+      // The container holds the form elements and actions.
+      $form['user_login_container'] = array(
+        '#type' => 'container',
+        '#attributes' => array('class' => array('user-login-container')),
+      );
+
+      // Only add a close button to the block form.
+      if ($form_id === 'user_login_block') {
+        $form['user_login_container']['close']['#markup'] = '<div class="close-user-login"></div>';
+      }
+
+      $form['user_login_container']['name'] = $form['name'];
+      $form['user_login_container']['pass'] = $form['pass'];
+
+      if (!empty($user_signup_link)) {
+        $form['user_login_container']['sign_up_link'] = array(
+          '#markup' => l(t('Register as a user'), $user_signup_link, array('external' => TRUE)),
+          '#weight' => 9999,
+          '#prefix' => '<div class="sign-up-link">',
+          '#suffix' => '</div>',
+        );
+      }
+
+      // Wrap the submit button so it can get an icon.
+      $form['actions']['submit']['#prefix'] = '<div class="submit-button-with-icon"><div class="color-and-icon"></div>';
+      $form['actions']['submit']['#suffix'] = '</div>';
+
+      // Add NemID if the gatewayf module exists.
+      if (module_exists('ding_gatewayf')) {
+        $form['#attributes']['class'][] = 'has-nemid-login';
+        $block = module_invoke('ding_gatewayf', 'block_view', 'login');
+        $block->content['#link']['#text'] = t('Login with');
+        $form['actions']['nemid'] = array('#markup' => render($block->content));
+      }
+
+      $form['user_login_container']['actions'] = $form['actions'];
+
+      // Remove links and all the elements that are now in the elements
+      // container.
```

Comment:
Putting the comment on 1 line would exceed 80 characters, and cause a warning (Drupal.Files.LineLength.TooLong).